src/plugins/discover/public/services/discover_ebt_manager.ts

```diff
@@ -216,4 +267,30 @@ export class DiscoverEBTManager {
       filterOperation,
     });
   }
+
+  public trackContextualProfileResolvedEvent({
+    contextLevel,
+    profileId,
+  }: {
+    contextLevel: ContextualProfileLevel;
+    profileId: string;
+  }) {
+    if (!this.reportEvent) {
+      return;
+    }
+
+    if (this.lastResolvedContextProfiles[contextLevel] === profileId) {
+      // avoid sending duplicate events to EBT
+      return;
+    }
```

Comment:
**non-blocking(unrelated)**: just a minor consideration I thought about after seeing this, AFAIK we don't have any runtime check on profiled being unique across the registered profiles, nor we enforce a literal convention with TS.

This leaves space for some corner cases such as solution registering profiles at the same context level that might have the same id (human error level).
For example, in obs we can create a logs profile with id "logs-data-source-profile" and then the security folks might to the same in a new profile, creating a more specific logic and without knowing the same id is already taken.
This would work well since the resolution logic might be different, but would break this telemetry since the Id might be the same right?

It's a really remote scenario, but we could be defensive and just add a check that watch out for registered profileIds when a profile is registered, since they are added sequentially.

Reply:
Yes, good point @tonyghiani 

Let's create a ticket so we can address this separately and ensure unique profile ids moving forward.